lib/src/logger.dart

```diff
@@ -27,21 +27,29 @@ class Logger {
 
   /// The full name of this logger, which includes the parent's full name.
   String get fullName =>
-      (parent == null || parent.name == '') ? name : '${parent.fullName}.$name';
+      parent?.name.isNotEmpty == true ? '${parent!.fullName}.$name' : name;
```

Comment:
[nit] the style guide recommends `?? false` over `== true` for nullable booleans.